pkg/cloud/vsphere/actuators/machine/actuator.go

```
@@ -0,0 +1,231 @@
+	coreClient corev1.CoreV1Interface) *Actuator {
+
+	return &Actuator{
+		client:     client,
+		coreClient: coreClient,
+	}
+}
+
+// Create creates a new machine.
+func (a *Actuator) Create(
+	parentCtx goctx.Context,
+	cluster *clusterv1.Cluster,
+	machine *clusterv1.Machine) (result error) {
+
+	ctx, err := context.NewMachineContext(
+		&context.MachineContextParams{
+			ClusterContextParams: context.ClusterContextParams{
+				Context:    parentCtx,
+				Cluster:    cluster,
+				Client:     a.client,
+				CoreClient: a.coreClient,
+				Logger:     klogr.New().WithName("[machine-actuator]"),
+			},
+			Machine: machine,
+		})
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if result == nil {
+			record.Eventf(ctx.Machine, "CreateSuccess", "created machine %q", ctx)
+		} else {
+			record.Warnf(ctx.Machine, "CreateFailure", "failed to create machine %q: %v", ctx, result)
+		}
+	}()
+
+	machineRole := ctx.Role()
+	if machineRole == "" {
+		return errors.Errorf("unable to get machine role while creating machine %q", ctx)
+	}
+
+	ctx.Logger.V(2).Info("creating machine", "role", machineRole)
+	defer ctx.Patch()
+
+	if !ctx.ClusterConfig.CAKeyPair.HasCertAndKey() {
+		ctx.Logger.V(2).Info("cluster config is missing pki toolchain, requeue machine")
+		return &clustererr.RequeueAfterError{RequeueAfter: constants.RequeueAfterSeconds}
+	}
+
+	controlPlaneMachines, err := ctx.GetControlPlaneMachines()
+	if err != nil {
+		return errors.Wrapf(err, "unable to get control plane machines while creating machine %q", ctx)
+	}
+
+	// Init the control plane by creating this machine.
+	if machineRole == context.ControlPlaneRole && len(controlPlaneMachines) == 1 {
+		if err := govmomi.Create(ctx, ""); err != nil {
+			return errors.Wrapf(err, "failed to create machine as initial member of the control plane %q", ctx)
+		}
+		return nil
+	}
+
+	// Join the existing cluster.
+	online, _, _ := kubeclient.GetControlPlaneStatus(ctx.ClusterContext)
```

check err

---

Hi @andrewsykim,

The error is ignored because the reaction is to requeue. Since the reaction doesn't depend on the error, only whether `online` is `true` or `false` (and will be `false` if there *is* an error), the error can be safely ignored in this case.